dace/codegen/CMakeLists.txt

```diff
@@ -532,6 +532,8 @@ if(DACE_ENABLE_INTELFPGA)
     string(REGEX REPLACE "kernel_(.+).cl" "\\1" DACE_KERNEL_NAME "${DACE_KERNEL_NAME}")
     set(DACE_AOC_KERNEL_FILES ${DACE_AOC_KERNEL_FILES} ${DACE_KERNEL_FILE})
 
+    # The output AOCX file will be named as the last OpenCL file given in input to the compiler
+    get_filename_component(DACE_AOC_OUTPUT_FILE ${DACE_KERNEL_FILE} NAME_WE)
```

Comment:
What about using the name of the top-level SDFG instead? Seems a bit more intuitive

Reply:
If you call the compiler `aoc file1.cl file2.cl file3.cl`, the outputput will be named `file3.aocx`

Reply:
You can probably just use `${DACE_PROGRAM_NAME}`

Reply:
No, that does not work. CL files are not named using program name but using the state names.
For example, in the test, the program name is "multiple_kernels", but the two files (states) are "fpga_state_0.cl" and "fpga_state_1.cl"

Reply:
Yes, but aren't we talking about the name of the _output_ file? There is only one of them, can that not use the program name?

Reply:
That is done later, and, yes, is done using the Program name

```
COMMAND mv ${DACE_AOC_OUTPUT_FILE}.aocx  ${DACE_PROGRAM_NAME}_emulator.aocx
```

Reply:
Why this extra step of moving it instead of using the output name directly?

Reply:
Because you can not use `-o` if you have multiple input files

```
$ aoc fpga_state_0.cl fpga_state_1.cl -o out.aocx
Error: Cannot specify -o with multiple input files
```

Reply:
Ahaaaa I see: please write this exact problem in more detail in the comment in the CMake script, then it will be clear to the next person :-)